src/posix/platform/radio_spinel.cpp

```diff
@@ -1365,7 +1365,7 @@ uint32_t RadioSpinel::GetRadioChannelMask(bool aPreferred)
 {
     uint8_t        maskBuffer[kChannelMaskBufferSize];
     otError        error       = OT_ERROR_NONE;
-    uint32_t       channelMask = 0;
+    uint32_t       channelMask = OT_RADIO_2P4GHZ_OQPSK_CHANNEL_MASK;
```

Comment:
We should honor what we get from RCP? 
Why would it fail to respond and if it does, should we even continue operation? (our RCP is not responding to a required request)

Reply:
It fails when we run a old version rcp which doesn't support the feature. Updated to use `SuccessOrDie()`.

Reply:
We need to define a clear rule/policy related to backward compatibility in RCP/Posix-app and adhere to it. 
Seems like we currently expect RCP and posix-app to be updated at the same time